### PR TITLE
SST sheet parser - fix.

### PR DESCRIPTION
**Pull request by bgrochal**
Things done:
- SST parsing fixed: increasing SST counter on empty value inside the `<si>` tag
- minor code improvements
- `0.0.8-AV2` version

### CODE REVIEW

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 1]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
-            case Characters(text) if si =>
-              lastContent match {
-                case Some(t) => t.append(text)
-                case None    => lastContent = Some(new StringBuilder().append(text))
-              }
-              Nil
-            case _ => Nil
-          }
+
+        data: ParseEvent => data match {
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: This line can just contain the opening bracket.

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 2]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
+          case StartElement("si", _, _, _, _) =>
+            si = true
+            Nil
+          case EndElement("si") =>
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: You might want to extract this local name to a constant.

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 3]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
         var count                              = 0
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: What does this count? (naming)

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 4]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
         var si                                 = false
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: Is that in Spanish?

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 5]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
-              Nil
-            case _ => Nil
-          }
+
+        data: ParseEvent => data match {
+          case StartElement("si", _, _, _, _) =>
+            si = true
+            Nil
+          case EndElement("si") =>
+            val sst = lastContent match {
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: Is this actually an SST (shared string table)?

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 6]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
+              case None =>
+                Nil
+            }
+            si = false
+            count += 1
+            sst
+          case Characters(text) if si =>
+            lastContent match {
+              case Some(builder) => builder.append(text)
+              case None => lastContent = Some(new StringBuilder().append(text))
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: Could be more direct `Some(new StringBuilder(text))`

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala [hunk 7]
```diff
@@ -55,30 +55,30 @@ object SstStreamer {
...
+            }
+            Nil
+          case _ => Nil
+        }
       })
```
Comment (ddworak) [inline src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: You can deduplicate the braces if you like.

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala (hunk 3 above)
Comment (bgrochal) [reply src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: In fact, this is a shared string index.

File: src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala (hunk 4 above)
Comment (bgrochal) [reply src/main/scala/akka/stream/alpakka/xlsx/SstStreamer.scala]: No, `si` stands for "string item". More: https://docs.microsoft.com/en-us/office/open-xml/working-with-the-shared-string-table.

**Comment by bgrochal** [comment]
Updated, please review @ddworak.

**Pull request merged**